docs/index.bs

```diff
@@ -3177,6 +3245,19 @@ spec: storage; urlPrefix: https://storage.spec.whatwg.org/
       1. Return |response|.
   </section>
 
+  <section algorithm>
+    <h3 id="get-router-source-algorithm"><dfn>GetRouterSource</dfn></h3>
+      : Input
+      :: |rules|, a list of {{RouterRule}}
+      :: |request|, a [=/request=]
+      : Output
+      :: {{RouterSourceEnum}} or null
+
+      1. For each |rule| in |rules|:
+          1. If |request|'s [=request/URL=] [=urlpattern/matches=] |rule|'s {{urlPattern}}, then returns |rule|'s {{RouterRule/source}}.
```

Comment:
rule's urlPattern (better written as `|rule|["{{RouterRule/condition}}"]["{{RouterCondition/urlPattern}}"]`) is a string. https://wicg.github.io/urlpattern/#match takes a `URLPattern` object, not a string. This is related to my issue above about needing to store the parsing result somewhere.

(Probably, you will need a separate hierarchy of parsed router rule [structs](https://infra.spec.whatwg.org/#structs), separate from the raw `RouterRule` dictionaries.)

Reply:
I tried to clarify that part.  I made RouterCondition/urlPattern to have an associated component.